### PR TITLE
add arraymerge tdigest function

### PR DESCRIPTION
**Pull request by ampampamp**
## Description
Adding a scalar function to merge T-digests from an input array.

## Motivation and Context
Existing issue: https://github.com/prestodb/presto/issues/18402

The aggregate function [`MERGE`](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/aggregation/MergeTDigestFunction.java) has existed for digests, but never the scalar equivalent.

## Impact
No impact, net-new function.

## Test Plan
Added unit tests.

Built with:
```
./mvnw clean install -Dtest="TestTDigest,TestTDigestFunctions" -Dmaven.javadoc.skip=true -T1C -fn -pl presto-main
```

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

== RELEASE NOTES ==

General Changes
* Adding the function `ARRAY_MERGE`.
  * Usage: `ARRAY_MERGE(CAST(ARRAY[digest1, digest2, ...] AS ARRAY(TDIGEST(DOUBLE))))`

### CODE REVIEW

**Comment by ampampamp** [comment]
Signed CLA.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/TDigestFunctions.java [hunk 1]
```diff
@@ -181,4 +182,31 @@ public static Slice constructTDigest(
...
+        }
+        Slice initialSlice = input.getSlice(firstNonNullIndex, 0, input.getSliceLength(firstNonNullIndex));
+        output = createTDigest(initialSlice);
+        for (int i = firstNonNullIndex + 1; i < input.getPositionCount(); i++) {
+            if (input.isNull(i)) {
+                continue;
+            }
+            output.merge(createTDigest(input.getSlice(i, 0, input.getSliceLength(i))));
+        }
+        return output.serialize();
```
[the remark this answers is not in the document]
Comment (tdcmeehan) [reply presto-main/src/main/java/com/facebook/presto/operator/scalar/TDigestFunctions.java:210]: Why treat the first null separately from the rest?

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestTDigestFunctions.java [hunk 2]
```diff
@@ -665,6 +665,79 @@ public void testConstructTDigestInverse()
...
+        SqlVarbinary sqlVarbinary = functionAssertions.selectSingleValue(
+                format("array_merge(cast(array[%s, %s] as array(tdigest(double))))",
+                        toSqlString(digest1),
+                        toSqlString(digest2)),
+                TDIGEST_DOUBLE,
+                SqlVarbinary.class);
+        digest1.merge(digest2);
+        assertEquals(sqlVarbinary, new SqlVarbinary(digest1.serialize().getBytes()));
+    }
+    @Test
```
Comment (tdcmeehan) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestTDigestFunctions.java:699]: Please add new lines in between test cases

**Comment by tdcmeehan** [comment]
Please address checkstyle checks

**Pull request closed without merging**